src/people.rs

```diff
@@ -427,15 +547,110 @@ mod test {
 
         let flag = Rc::new(RefCell::new(false));
         let flag_clone = flag.clone();
-        context.subscribe_to_event(
-            move |_context, _event: PersonPropertyChangeEvent<RunningShoes>| {
+        context.subscribe_to_person_property_changed(
+            RunningShoes,
+            move |_context, _person, _current, _prev| {
                 *flag_clone.borrow_mut() = true;
             },
         );
         let person_id = context.add_person();
-        // Innitializer wasn't called, so don't fire an event
+        // Initializer wasn't called, so don't fire an event
         context.set_person_property(person_id, RunningShoes, 42);
         context.execute();
         assert!(!*flag.borrow());
     }
+
+    #[test]
+    fn derived_property_returns_correct_values() {
+        let mut context = Context::new();
+        define_derived_person_property!(MastersRunner, bool, [Age, IsRunner], |age, is_runner| age
+            >= 40
+            && is_runner);
+
+        let paula = context.add_person();
+        context.set_person_property(paula, Age, 50);
+        context.set_person_property(paula, IsRunner, true);
+
+        let colleen = context.add_person();
+        context.set_person_property(colleen, Age, 31);
+        context.set_person_property(colleen, IsRunner, true);
+
+        assert!(context.get_person_property(paula, MastersRunner),);
+        assert!(!context.get_person_property(colleen, MastersRunner),);
+    }
+
+    #[test]
+    fn derived_property_initializes() {
+        let mut context = Context::new();
+        define_person_property!(Friendly, bool, |_context, _person_id| true);
+        define_person_property!(IsHuman, bool, |_context, _person_id| true);
+        define_derived_person_property!(
+            Likeable,
+            bool,
+            [Friendly, IsHuman],
+            |friendly, is_human| { friendly && is_human }
+        );
+
+        let person = context.add_person();
+
+        assert!(context.get_person_property(person, Likeable));
+    }
+
+    #[test]
+    fn derived_property_change_event() {
+        let mut context = Context::new();
+        define_derived_person_property!(MastersRunner, bool, [Age, IsRunner], |age, is_runner| age
+            >= 40
+            && is_runner);
+
+        let person = context.add_person();
+        context.set_person_property(person, Age, 50);
+        context.set_person_property(person, IsRunner, false);
```

Comment:
You need to stop people from externally calling set_person_property() on derived. I think I would do this by having a set_person_property_int() that is private but called by set_person_property()